dotnet-client-libraries/ParsingHtmlWithAngleSharp/Tests/ParsingHtmlWithAngleSharpUnitTests.cs

```diff
@@ -0,0 +1,290 @@
+    {
+        var booksCatalogUrl = "https://books.toscrape.com/";
+
+        var config = Configuration.Default
+            .WithDefaultLoader()
+            .WithJs();
+        var context = BrowsingContext.New(config);
+
+        var document = await context.OpenAsync(new Url(booksCatalogUrl));
+
+        var booksSection = document.QuerySelector<IHtmlElement>("div.page_inner section")!;
+        var bookInfoArticles = booksSection
+            .QuerySelectorAll<IHtmlElement>("li > article.product_pod")
+            .ToCollection();
+
+        var books = bookInfoArticles.Select(ToBook).ToList();
+        Console.WriteLine(books.Count()); // 20
+
+        Assert.Equal(20, books.Count);
+        Assert.Equal("A Light in the Attic", books[0].Title);
+        Assert.Equal("Starving Hearts (Triangular Trade Trilogy, #1)", books[10].Title);
+        Assert.Equal("It's Only the Himalayas", books[^1].Title);
+    }
+
+    [Fact]
+    public async Task WhenParsingHtmlFromString_ThenDocumentIsCreatedAndDomManipulationChangesDomAsExpected()
+    {
+        var config = Configuration.Default
+            .WithDefaultLoader()
+            .WithJs();
+
+        var context = BrowsingContext.New(config);
+        var document = await context.OpenAsync(req => req.Content(Html));
+
+        var paragraphElement = document.CreateElement("p");
+        // OR
+        paragraphElement = document.CreateElement<IHtmlParagraphElement>();
+        paragraphElement.TextContent = "This is a new paragraph.";
+
+        document.Body.AppendChild(paragraphElement);
+
+        Assert.IsAssignableFrom<IHtmlParagraphElement>(document.Body.Children[^1]);
+
+        var ulElement = document.QuerySelector<IHtmlUnorderedListElement>("ul#list")!;
+        var blueLiElement = ulElement.QuerySelector<IHtmlListItemElement>("li.blue")!;
+
+        blueLiElement.Remove();
+        // OR
+        // ulElement.RemoveChild(blueLiElement);
+
+        Assert.Equal(2, ulElement.Children.Length);
+
+        var article = document.QuerySelector<IElement>("article#a1")!;
+        article.TextContent = "New article content";
+        article.InnerHtml = "New article content. <br /> Second article row.";
+
+        article.ClassList.Add("small-article");
+        article.Id = "news-article";
+
+        article.SetAttribute("data-category", "news");
+
+        Assert.Contains(article.Attributes, attr => attr.Name == "data-category");
+
+        article.RemoveAttribute("data-category");
+
+        Assert.Equal("New article content. <br> Second article row.", article.InnerHtml);
+        Assert.IsAssignableFrom<IHtmlBreakRowElement>(article.Children[0]);
+        Assert.Equal(1, article.ClassList.Length);
+        Assert.Equal("small-article", article.ClassList[0]);
+        Assert.NotEmpty(article.Id);
+        Assert.DoesNotContain(article.Attributes, attr => attr.Name == "data-category");
+
+        var writer = new StringWriter();
+        Console.SetOut(writer);
```

Comment:
If we don't need these, we can remove them. Including the other `Console` related code.

Reply:
Is here the console logging the problem? Because in these tests, I'm logging into an in-memory `StringWriter` and not the actual console ...

Reply:
@vnikolaew 
This case might not apply. But there is another instance where the `Console.WriteLine` is redundant.